### PR TITLE
[3D] Add config3d property

### PR DESCRIPTION
**Pull request by oterral**
This PR is based on #2590. It adds a config3d property to the layer which contains what we nee to use the layer in 3d.

@gjn is it what you had in mind when we talk about client modification of the layersconfig?

### CODE REVIEW

**Comment by oterral** [comment]
@gjn rebased , ready for  a first review

File: src/components/map/MapService.js [hunk 1]
```diff
@@ -820,6 +839,37 @@ goog.require('ga_urlutils_service');
...
           return $http.get(url).then(function(response) {
+            // Live modfiications for 3d test
+            if (response.data) {
+              var ids = [
+                'ch.swisstopo.swisstlm3d-karte-farbe',
+                'ch.swisstopo.swissimage-product'
+              ];
+              angular.forEach(ids, function(id) {
+                if (response.data[id]) {
+                  response.data[id].config3d = id + '_3d';
```
Comment (gjn) [inline src/components/map/MapService.js]: 3d id's should be in array, no?

File: src/components/map/MapService.js [hunk 2]
```diff
@@ -820,6 +839,37 @@ goog.require('ga_urlutils_service');
...
+              response.data['ch.swisstopo.swissimage-product_3d'] = {
+                type: 'wmts',
+                url: '//wmts.geo.admin.ch/1.0.0/' +
+                    '00_todelete_ch.swisstopo.swissimage-product/default/' +
+                    '20151231_v03/{TileMatrixSet}/{TileMatrix}/{TileRow}/{TileCol}.jpeg'
+              };
+              // Terain
+              response.data['ch.swisstopo.terrain.3d'] = {
+                type: 'terrain',
+                serverLayerName: 'ch.swisstopo.terrain.3d',
```
Comment (gjn) [inline src/components/map/MapService.js]: Should we take default terrain here (coming from index via Makefile)?

Comment (gjn) [inline src/components/map/MapService.js]: Ignore this.

File: src/components/map/MapService.js (hunk 1 above)
Comment (oterral) [inline src/components/map/MapService.js]: it depends. I thought, the id can refers to an aggregate layer instead of a list of ids

Comment (gjn) [inline src/components/map/MapService.js]: In the spec, all layers are in an array: https://docs.google.com/document/d/1vtSbZhEwL8W8rmT6iLKlrDsMuccpKyoORqWEPGbUjb4/edit

Comment (oterral) [inline src/components/map/MapService.js]: Yes i fix that.

**Comment by oterral** [comment]
PR Updated

File: src/index.mako.html [hunk 3]
```diff
@@ -718,9 +719,15 @@
...
         module.config(function(gaLayersProvider, gaGlobalOptions) {
-          gaLayersProvider.wmtsGetTileUrlTemplate =
-              '//wmts{5-9}.geo.admin.ch/1.0.0/{Layer}/default/{Time}/21781/{TileMatrix}/{TileRow}/{TileCol}.{Format}';
-
+          gaLayersProvider.wmtsGetTileUrlTemplate = '//wmts{5-9}.geo.admin.ch/1.0.0/{Layer}/default/{Time}/21781/{TileMatrix}/{TileRow}/{TileCol}.{Format}';
+          
+          gaLayersProvider.wmtsMapProxyGetTileUrlTemplate =
+          '//mf-chsdi3.dev.bgdi.ch/mom_fix_1594/1.0.0/{Layer}/default/{Time}/{TileMatrixSet}/{TileMatrix}/{TileCol}/{TileRow}.{Format}';
+          
+          gaLayersProvider.terrainTileUrlTemplate = '//3d.dev.bgdi.ch/1.0.0/{Layer}/default/{Time}/4326';
```
Comment (gjn) [inline src/index.mako.html]: We should use official address `3d.geo.admin.ch`. It's working again.

File: ol3-cesium-plugin/garastersynchronizer.js [hunk 4]
```diff
@@ -34,38 +34,9 @@ ga.GaRasterSynchronizer.prototype.convertLayerToCesiumImagery =
...
-  } else if (source instanceof ol.source.TileWMS ||
-             source instanceof ol.source.ImageWMS) {
-    provider = new Cesium.WebMapServiceImageryProvider({
-      url: source.getUrls ? source.getUrls()[0] :
-                            /** @type {string} */ (source.getUrl()),
-      layers: layerId,
-      parameters: {format:'image/png'}
-    });
+  var getProvider = olLayer['getCesiumImageryProvider'];
+  if (getProvider) {
```
Comment (gberaudo) [inline ol3-cesium-plugin/garastersynchronizer.js]: Is it possible that `getProvider` be undefined?

**Comment by gjn** [comment]
This LGTM. Tested as well and works well.

File: ol3-cesium-plugin/garastersynchronizer.js (hunk 4 above)
Comment (oterral) [inline ol3-cesium-plugin/garastersynchronizer.js]: I don't think so.

Comment (oterral) [inline ol3-cesium-plugin/garastersynchronizer.js]: Ah yes it could be for Extenral WMS Layer.

Comment (gberaudo) [inline ol3-cesium-plugin/garastersynchronizer.js]: If not having a `getProvider` is possible, isn't it wrong to throw?

Comment (oterral) [inline ol3-cesium-plugin/garastersynchronizer.js]: Technically an external WMS could be displayed in 3d  if the WMS support 1.1.1 so it should have one. And yes should not be thrown if `getProvider`is not defined.
I fix.

**Comment by oterral** [comment]
It's good now. Do you other thing @gberaudo ?

**Comment by gberaudo** [comment]
@oterral, it looks good for me now.

**Comment by oterral** [comment]
ok thx.

**Pull request merged**